# See https://github.com/SynoCommunity/spksrc/issues/2662
# Remove the extension pgsql because the extension pgsql does not work with the
# PostgreSQL server from DSM. The Extension pdo_pgsql works perfectly.
--- adminer/drivers/pgsql.inc.php.orig
+++ adminer/drivers/pgsql.inc.php
@@ -6,155 +6,7 @@
 if (isset($_GET["pgsql"])) {
 	define('Adminer\DRIVER', "pgsql");
 
-	if (extension_loaded("pgsql") && $_GET["ext"] != "pdo") {
-		class PgsqlDb extends SqlDb {
-			public $extension = "PgSQL";
-			public $timeout = 0;
-			private $link, $string;
-
-			// no type declarations, this is passed to set_error_handler()
-			function _error($errno, $error) {
-				$this->error = preg_replace('~^[^:]*: ~', '', php_error_text($error));
-			}
-
-			function attach(array $server, string $username, string $password): string {
-				$db = adminer()->database();
-				set_error_handler(array($this, '_error'));
-				$port = $server["port"];
-				$host = ($server["host"] ?: $server["socket"]); // the socket is passed as the host directory
-				$this->string = "host='$host'" . ($port ? " port=$port" : "") . " user='" . addcslashes($username, "'\\") . "' password='" . addcslashes($password, "'\\") . "'";
-				$ssl = adminer()->connectSsl();
-				if (isset($ssl["mode"])) {
-					$this->string .= " sslmode=$ssl[mode]";
-				}
-				$this->link = @pg_connect("$this->string dbname='" . ($db != "" ? addcslashes($db, "'\\") : "postgres") . "'", PGSQL_CONNECT_FORCE_NEW);
-				if (!$this->link && $db != "") {
-					// try to connect directly with database for performance
-					$this->link = @pg_connect("$this->string dbname='postgres'", PGSQL_CONNECT_FORCE_NEW);
-				}
-				restore_error_handler();
-				if ($this->link) {
-					pg_set_client_encoding($this->link, "UTF8");
-				}
-				return ($this->link ? '' : $this->error);
-			}
-
-			function quote(string $string): string {
-				return (function_exists('pg_escape_literal')
-					? pg_escape_literal($this->link, $string) // available since PHP 5.4.4
-					: "'" . pg_escape_string($this->link, $string) . "'"
-				);
-			}
-
-			function value(?string $val, array $field): ?string {
-				return ($field["type"] == "bytea" && $val !== null ? pg_unescape_bytea($val) : $val);
-			}
-
-			function select_db(string $database) {
-				if ($database == pg_dbname($this->link)) { // the connection is to the database directly or to postgres if it failed
-					return true;
-				}
-				set_error_handler(array($this, '_error'));
-				$return = @pg_connect("$this->string dbname='" . addcslashes($database, "'\\") . "'", PGSQL_CONNECT_FORCE_NEW);
-				restore_error_handler();
-				if ($return) {
-					$this->link = $return;
-				}
-				return $return;
-			}
-
-			function close(): void {
-				$this->link = @pg_connect("$this->string dbname='postgres'");
-			}
-
-			function query(string $query, bool $unbuffered = false) {
-				if (self::$untrusted) {
-					// the extended protocol doesn't allow multiple commands, the transaction doesn't allow modifying data
-					$result = (@pg_query($this->link, "BEGIN READ ONLY")
-						? @pg_query_params($this->link, $query, array())
-						: false
-					);
-				} else {
-					$result = @pg_query($this->link, $query);
-				}
-				$this->error = "";
-				if (!$result) {
-					$this->error = pg_last_error($this->link);
-					$return = false;
-				} elseif (!pg_num_fields($result) && pg_result_status($result) != PGSQL_TUPLES_OK) { // SELECT without columns returns rows
-					$this->affected_rows = pg_affected_rows($result);
-					$return = true;
-				} else {
-					$return = new Result($result);
-				}
-				if (self::$untrusted) {
-					@pg_query($this->link, "COMMIT"); // rollbacks the transaction if the query failed
-				}
-				if ($this->timeout) {
-					$this->timeout = 0;
-					$this->query("RESET statement_timeout");
-				}
-				return $return;
-			}
-
-			function warnings() {
-				if (PHP_VERSION_ID >= 70100) {
-					$return = implode("\n", pg_last_notice($this->link, PGSQL_NOTICE_ALL));
-					pg_last_notice($this->link, PGSQL_NOTICE_CLEAR);
-				} else {
-					$return = pg_last_notice($this->link);
-				}
-				return nl_br(h($return));
-			}
-
-			function inTransaction(): bool {
-				$status = pg_transaction_status($this->link);
-				return $status == PGSQL_TRANSACTION_INTRANS || $status == PGSQL_TRANSACTION_INERROR; // not PGSQL_TRANSACTION_UNKNOWN - the connection is bad
-			}
-
-			/** Copy from array into a table
-			* @param list<string> $rows
-			*/
-			function copyFrom(string $table, array $rows): bool {
-				$this->error = '';
-				set_error_handler(function (int $errno, string $error): bool {
-					$this->error = php_error_text($error);
-					return true;
-				});
-				$return = pg_copy_from($this->link, $table, $rows);
-				restore_error_handler();
-				return $return;
-			}
-		}
-
-		class Result {
-			public $num_rows;
-			private $result, $offset = 0;
-
-			function __construct($result) {
-				$this->result = $result;
-				$this->num_rows = pg_num_rows($result);
-			}
-
-			function fetch_assoc() {
-				return pg_fetch_assoc($this->result);
-			}
-
-			function fetch_row() {
-				return pg_fetch_row($this->result);
-			}
-
-			function fetch_field(): \stdClass {
-				$column = $this->offset++;
-				$return = new \stdClass;
-				$return->orgtable = pg_field_table($this->result, $column);
-				$return->name = pg_field_name($this->result, $column);
-				$return->native_type = pg_field_type($this->result, $column);
-				return $return;
-			}
-		}
-
-	} elseif (extension_loaded("pdo_pgsql")) {
+	if (extension_loaded("pdo_pgsql")) {
 		class PgsqlDb extends PdoDb {
 			public $extension = "PDO_PgSQL";
 			public $timeout = 0;
